File: Day2/cubeConundrum.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>


#define  RED_COUNT = 12;
#define  GREEN_COUNT = 13;
#define  BLUE_COUNT = 14;


int main(int argc, char *argv[]) {


    for (int i = 1; i < argc; i++) {
        char count[10];
        char color[10];

        if (i < argc - 1) {
            strcpy(count, argv[i]);
            strcpy(color, argv[i + 1]);
            i++;

            if (strcmp(color, "red,") == 0) {
                RED_COUNT -= atoi(count);
            } else if (strcmp(color, "blue,") == 0) {
                BLUE_COUNT -= atoi(count);
            } else if (strcmp(color, "green,") == 0) {
                GREEN_COUNT -= atoi(count);
            }
        }
    }

    if (RED_COUNT < 0 || BLUE_COUNT < 0 || GREEN_COUNT < 0) {
        printf("This game is not possible\n");
    } else {
        printf("This game is possible\n");
    }

    return 0;
}

File: Day2/cubeConundrum.py
def cubeConundrum(game_input):

    games = game_input.split('; ')
    for game in games:
        red_count = 12
        green_count = 13
        blue_count = 14
        cubes = game.split(', ')
        for cube in cubes:
            count, color = cube.split(' ')
            count = int(count)

            if color == 'red':
                red_count -= count
            elif color == 'green':
                green_count -= count
            elif color == 'blue':
                blue_count -= count

        if red_count < 0 or blue_count < 0 or green_count < 0 :
            return False

    return True



def tryCubeConundrum():
    i = 1
    j = 1
    with open('Day2/games.txt', 'r') as file:
        for line in file:
            is_possible = cubeConundrum(line)
            if is_possible:
                print("game",i+j-1 ,"is possible")
                i = i+1
            else:
                print("game",i+j-1 ,"is not possible")
                j = j+1

    print("games possible : ",i, " games impossible : ", j)
tryCubeConundrum()
